trestle/core/catalog_interface.py

```diff
@@ -650,85 +651,200 @@ def write_catalog_as_markdown(self, context: ControlContext, part_id_map: Dict[s
+                if new_context.overwrite_header_values:
+                    # update the control params with new values
+                    for key, value in new_context.yaml_header[const.SET_PARAMS_TAG].items():
+                        if key in control_param_dict:
+                            set_param_dict[key] = value
+                else:
+                    # update the control params with any values in yaml header not set in control
+                    # need to maintain order in the set_param_dict
+                    for key, value in new_context.yaml_header[const.SET_PARAMS_TAG].items():
+                        if key in control_param_dict and key not in set_param_dict:
+                            set_param_dict[key] = value
+                new_context.yaml_header[const.SET_PARAMS_TAG] = set_param_dict
+            elif const.SET_PARAMS_TAG in new_context.yaml_header:
+                # need to cull any params that are not in control
+                pop_list: List[str] = []
+                for key in new_context.yaml_header[const.SET_PARAMS_TAG].keys():
+                    if key not in control_param_dict:
+                        pop_list.append(key)
+                for pop in pop_list:
+                    new_context.yaml_header[const.SET_PARAMS_TAG].pop(pop)
+
+            new_context.required_sections = required_section_list
+            new_context.allowed_sections = allowed_section_list
+            found_control_alters = as_filtered_list(found_alters, lambda a: a.control_id == control.id)
+
+            self._write_control_into_dir(new_context, control, part_id_map, found_control_alters)
+
+    def write_catalog_as_ssp_markdown(self, context: ControlContext, part_id_map: Dict[str, Dict[str, str]]) -> None:
+        """Write out the catalog as component markdown."""
+        # in component mode get rules, params, and param values from the current control_implementation
+        # the catalog is written out in pieces per control_imp
+        allowed_section_list = context.allowed_sections.split(',') if context.allowed_sections else []
+
+        # write out the controls
+        for control in self.get_all_controls_from_catalog(True):
+            # here we do special handling of how set-parameters merge with the yaml header
+            new_context = ControlContext.clone(context)
+            new_context.allowed_sections = allowed_section_list
+
+            self._write_control_into_dir(new_context, control, part_id_map, [])
+
+    def write_catalog_as_component_markdown(
+        self, context: ControlContext, part_id_map: Dict[str, Dict[str, str]]
+    ) -> None:
+        """Write out the catalog as component markdown."""
+
+        def _update_values(set_param: comp.SetParameter) -> None:
+            values = ', '.join([v.__root__ for v in as_list(set_param.values)])
+            if set_param.param_id in control_param_dict:
+                control_param_dict[set_param.param_id] = values
```

Comment:
control_param_dict contains str:Parameter but now it is string... I think this will fail further down the code when you call `ControlInterface.param_to_str(param)`

Reply:
fixed